<launch>
    <arg name="ns" default="robot_0"/>
    <arg name="robot_path" value="(find go2_description)"/>
    <arg name="dollar" value="$"/>

    <arg name="initial_x" default="0.0"/>
    <arg name="initial_y" default="0.0"/>
    <arg name="initial_z" default="0.5"/>

    <arg name="auto_fixedstand" default="true"/>
    <arg name="auto_fixedstand_delay_s" default="10.0"/>
    <arg name="auto_move_base" default="true"/>
    <arg name="auto_move_base_delay_s" default="3.0"/>
    <arg name="control_yaml" default="robot_control.yaml"/>

    <!-- Debug mode will hung up the robot, use "true" or "false" to switch it. -->
    <arg name="user_debug" default="false"/>

    <!-- Namespaced robot_name for nodes that don't use private params -->
    <!-- Load the URDF into the ROS Parameter Server -->

        <param name="robot_description" command="$(find xacro)/xacro --inorder '$(arg dollar)$(arg robot_path)/xacro/robot.xacro' DEBUG:=$(arg user_debug)"/>

    <group ns="$(arg ns)">
        <param name="tf_prefix" value="$(arg ns)/go2_gazebo"/>
        <param name="robot_description" command="$(find xacro)/xacro --inorder '$(arg dollar)$(arg robot_path)/xacro/robot.xacro' 
        DEBUG:=$(arg user_debug) --robot_namespace:=/$(arg ns)/go2_gazebo"/>

        <rosparam file="$(arg dollar)$(arg robot_path)/config/robot_control.yaml" command="load"/>


        <node pkg="gazebo_ros" type="spawn_model" name="urdf_spawner" args="-urdf -x $(arg initial_x) -y $(arg initial_y) -z $(arg initial_z)
                -model $(arg ns) -param robot_description -unpause" output="screen"/>

        <node pkg="controller_manager" type="spawner" name="controller_spawner" output="screen" ns="go2_gazebo" args="joint_state_controller 
                FL_hip_controller FL_thigh_controller FL_calf_controller
                FR_hip_controller FR_thigh_controller FR_calf_controller
                RL_hip_controller RL_thigh_controller RL_calf_controller
                RR_hip_controller RR_thigh_controller RR_calf_controller"/>

        <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" respawn="false" output="screen">
            <remap from="/$(arg ns)/joint_states" to="go2_gazebo/joint_states"/>
        </node>

        <node pkg="unitree_guide" type="junior_ctrl" name="junior_ctrl" output="screen">
            <param name="robot_name" value="$(arg ns)/go2"/>
            <param name="auto_fixedstand" value="$(arg auto_fixedstand)"/>
            <param name="auto_fixedstand_delay_s" value="$(arg auto_fixedstand_delay_s)"/>
            <param name="auto_move_base" value="$(arg auto_move_base)"/>
            <param name="auto_move_base_delay_s" value="$(arg auto_move_base_delay_s)"/>
        </node>

    </group>

    <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="static_odom_to_base" args="0 0 0 0 0 0 /$(arg ns)/go2_gazebo/odom /$(arg ns)/go2_gazebo/base"/> -->


</launch>